third-party/realdds/src/dds-device-impl.cpp

```diff
@@ -494,11 +499,15 @@ void dds_device::impl::create_control_writer()
 }
 
 
-void dds_device::impl::on_device_header( nlohmann::json const & j )
+void dds_device::impl::on_device_header( nlohmann::json const & j, eprosima::fastdds::dds::SampleInfo const & sample )
 {
     if( _state != state_t::WAIT_FOR_DEVICE_HEADER )
         return;
 
+    // The server GUID is the server's notification writer's GUID -- that way, we can easily associate all notifications
+    // with a server.
+    eprosima::fastrtps::rtps::iHandle2GUID( _server_guid, sample.publication_handle );
```

Comment:
I don't like passing unused parameter to all notigication handling functions just for this line. Can we get the info some other way? Maybe in on_data_available

Reply:
Unfortunately, this is the API given to us by FastDDS. The same one is used in the device-watcher.
And `_server_guid` is not uninitialized. The default ctor for a Guid_t sets it to all zeroes.

And "passing the unused parameter to all notification handling functions", you mean the sample info? It's up to the handler to decide whether to use it or not. It can certainly be used, just like it's used for control-reply handling on the server side. It has valuable information, like timestamps, sequence number, etc..

Reply:
OK